2020-04-04 18:25:25.967  INFO  28064 --- [restartedMain] application.MessageApplication           : Starting MessageApplication on GADs-MacBook-Pro-4.local with PID 28064 (/Applications/www/ttu-erp-java-monolith/srms-springboot/target/classes started by gadoo in /Applications/www/ttu-erp-java-monolith/srms-springboot)
 2020-04-04 18:25:25.980  INFO  28064 --- [restartedMain] application.MessageApplication           : No active profile set, falling back to default profiles: default
 2020-04-04 18:25:26.234  INFO  28064 --- [restartedMain] .e.DevToolsPropertyDefaultsPostProcessor : Devtools property defaults active! Set 'spring.devtools.add-properties' to 'false' to disable
 2020-04-04 18:25:26.234  INFO  28064 --- [restartedMain] .e.DevToolsPropertyDefaultsPostProcessor : For additional web related logging consider setting the 'logging.level.web' property to 'DEBUG'
 2020-04-04 18:25:26.929  INFO  28064 --- [restartedMain] .s.d.r.c.RepositoryConfigurationDelegate : Bootstrapping Spring Data repositories in DEFAULT mode.
 2020-04-04 18:25:26.989  INFO  28064 --- [restartedMain] .s.d.r.c.RepositoryConfigurationDelegate : Finished Spring Data repository scanning in 55ms. Found 4 repository interfaces.
 2020-04-04 18:25:26.995  INFO  28064 --- [restartedMain] .s.d.r.c.RepositoryConfigurationDelegate : Bootstrapping Spring Data repositories in DEFAULT mode.
 2020-04-04 18:25:27.001  INFO  28064 --- [restartedMain] .s.d.r.c.RepositoryConfigurationDelegate : Finished Spring Data repository scanning in 5ms. Found 0 repository interfaces.
 2020-04-04 18:25:27.611  INFO  28064 --- [restartedMain] o.s.b.w.embedded.tomcat.TomcatWebServer  : Tomcat initialized with port(s): 9090 (http)
 2020-04-04 18:25:27.621  INFO  28064 --- [restartedMain] o.apache.catalina.core.StandardService   : Starting service [Tomcat]
 2020-04-04 18:25:27.622  INFO  28064 --- [restartedMain] org.apache.catalina.core.StandardEngine  : Starting Servlet engine: [Apache Tomcat/9.0.27]
 2020-04-04 18:25:27.706  INFO  28064 --- [restartedMain] o.a.c.c.C.[.[localhost].[/ttu-erp]       : Initializing Spring embedded WebApplicationContext
 2020-04-04 18:25:27.706  INFO  28064 --- [restartedMain] o.s.web.context.ContextLoader            : Root WebApplicationContext: initialization completed in 1471 ms
 2020-04-04 18:25:27.915  INFO  28064 --- [restartedMain] org.mongodb.driver.cluster               : Cluster created with settings {hosts=[localhost:27017], mode=SINGLE, requiredClusterType=UNKNOWN, serverSelectionTimeout='30000 ms', maxWaitQueueSize=500}
 2020-04-04 18:25:27.967  WARN  28064 --- [cluster-ClusterId{value='5e88d1173ea5cc3c08d2aec1', description='null'}-localhost:27017] io.netty.util.internal.MacAddressUtil    : Failed to find a usable hardware address from the network interfaces; using random bytes: 9b:2e:b3:15:0d:f3:49:43
 2020-04-04 18:25:28.011  INFO  28064 --- [cluster-ClusterId{value='5e88d1173ea5cc3c08d2aec1', description='null'}-localhost:27017] org.mongodb.driver.cluster               : Exception in monitor thread while connecting to server localhost:27017

com.mongodb.MongoSocketOpenException: Exception opening socket
	at com.mongodb.connection.netty.NettyStream$OpenChannelFutureListener.operationComplete(NettyStream.java:410) ~[mongodb-driver-core-3.11.1.jar:na]
	at com.mongodb.connection.netty.NettyStream$OpenChannelFutureListener.operationComplete(NettyStream.java:385) ~[mongodb-driver-core-3.11.1.jar:na]
	at io.netty.util.concurrent.DefaultPromise.notifyListener0(DefaultPromise.java:577) ~[netty-common-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.util.concurrent.DefaultPromise.notifyListeners0(DefaultPromise.java:570) ~[netty-common-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.util.concurrent.DefaultPromise.notifyListenersNow(DefaultPromise.java:549) ~[netty-common-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.util.concurrent.DefaultPromise.notifyListeners(DefaultPromise.java:490) ~[netty-common-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.util.concurrent.DefaultPromise.setValue0(DefaultPromise.java:615) ~[netty-common-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.util.concurrent.DefaultPromise.setFailure0(DefaultPromise.java:608) ~[netty-common-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.util.concurrent.DefaultPromise.tryFailure(DefaultPromise.java:117) ~[netty-common-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.nio.AbstractNioChannel$AbstractNioUnsafe.fulfillConnectPromise(AbstractNioChannel.java:321) ~[netty-transport-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.nio.AbstractNioChannel$AbstractNioUnsafe.finishConnect(AbstractNioChannel.java:337) ~[netty-transport-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:688) ~[netty-transport-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:635) ~[netty-transport-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:552) ~[netty-transport-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:514) ~[netty-transport-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.util.concurrent.SingleThreadEventExecutor$6.run(SingleThreadEventExecutor.java:1044) ~[netty-common-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74) ~[netty-common-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30) ~[netty-common-4.1.42.Final.jar:4.1.42.Final]
	at java.base/java.lang.Thread.run(Thread.java:844) [na:na]
Caused by: io.netty.channel.AbstractChannel$AnnotatedConnectException: Connection refused: localhost/0:0:0:0:0:0:0:1:27017
Caused by: java.net.ConnectException: Connection refused
	at java.base/sun.nio.ch.SocketChannelImpl.checkConnect(Native Method) ~[na:na]
	at java.base/sun.nio.ch.SocketChannelImpl.finishConnect(SocketChannelImpl.java:717) ~[na:na]
	at io.netty.channel.socket.nio.NioSocketChannel.doFinishConnect(NioSocketChannel.java:327) ~[netty-transport-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.nio.AbstractNioChannel$AbstractNioUnsafe.finishConnect(AbstractNioChannel.java:334) ~[netty-transport-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:688) ~[netty-transport-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:635) ~[netty-transport-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:552) ~[netty-transport-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:514) ~[netty-transport-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.util.concurrent.SingleThreadEventExecutor$6.run(SingleThreadEventExecutor.java:1044) ~[netty-common-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74) ~[netty-common-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30) ~[netty-common-4.1.42.Final.jar:4.1.42.Final]
	at java.base/java.lang.Thread.run(Thread.java:844) [na:na]

 2020-04-04 18:25:28.087  INFO  28064 --- [restartedMain] org.mongodb.driver.cluster               : Cluster created with settings {hosts=[localhost:27017], mode=MULTIPLE, requiredClusterType=UNKNOWN, serverSelectionTimeout='30000 ms', maxWaitQueueSize=500}
 2020-04-04 18:25:28.088  INFO  28064 --- [restartedMain] org.mongodb.driver.cluster               : Adding discovered server localhost:27017 to client view of cluster
 2020-04-04 18:25:28.093  INFO  28064 --- [cluster-ClusterId{value='5e88d1183ea5cc3c08d2aec2', description='null'}-localhost:27017] org.mongodb.driver.cluster               : Exception in monitor thread while connecting to server localhost:27017

com.mongodb.MongoSocketOpenException: Exception opening socket
	at com.mongodb.internal.connection.SocketStream.open(SocketStream.java:70) ~[mongodb-driver-core-3.11.1.jar:na]
	at com.mongodb.internal.connection.InternalStreamConnection.open(InternalStreamConnection.java:128) ~[mongodb-driver-core-3.11.1.jar:na]
	at com.mongodb.internal.connection.DefaultServerMonitor$ServerMonitorRunnable.run(DefaultServerMonitor.java:117) ~[mongodb-driver-core-3.11.1.jar:na]
	at java.base/java.lang.Thread.run(Thread.java:844) [na:na]
Caused by: java.net.ConnectException: Connection refused (Connection refused)
	at java.base/java.net.PlainSocketImpl.socketConnect(Native Method) ~[na:na]
	at java.base/java.net.AbstractPlainSocketImpl.doConnect(AbstractPlainSocketImpl.java:400) ~[na:na]
	at java.base/java.net.AbstractPlainSocketImpl.connectToAddress(AbstractPlainSocketImpl.java:243) ~[na:na]
	at java.base/java.net.AbstractPlainSocketImpl.connect(AbstractPlainSocketImpl.java:225) ~[na:na]
	at java.base/java.net.SocksSocketImpl.connect(SocksSocketImpl.java:402) ~[na:na]
	at java.base/java.net.Socket.connect(Socket.java:591) ~[na:na]
	at com.mongodb.internal.connection.SocketStreamHelper.initialize(SocketStreamHelper.java:64) ~[mongodb-driver-core-3.11.1.jar:na]
	at com.mongodb.internal.connection.SocketStream.initializeSocket(SocketStream.java:79) ~[mongodb-driver-core-3.11.1.jar:na]
	at com.mongodb.internal.connection.SocketStream.open(SocketStream.java:65) ~[mongodb-driver-core-3.11.1.jar:na]
	... 3 common frames omitted

 2020-04-04 18:25:28.173  WARN  28064 --- [restartedMain] o.s.data.convert.CustomConversions       : Registering converter from class java.time.LocalDateTime to class java.time.Instant as reading converter although it doesn't convert from a store-supported type! You might wanna check you annotation setup at the converter implementation.
 2020-04-04 18:25:28.173  WARN  28064 --- [restartedMain] o.s.data.convert.CustomConversions       : Registering converter from class java.time.Instant to class java.time.LocalDateTime as reading converter although it doesn't convert from a store-supported type! You might wanna check you annotation setup at the converter implementation.
 2020-04-04 18:25:28.284  INFO  28064 --- [restartedMain] o.s.b.d.a.OptionalLiveReloadServer       : LiveReload server is running on port 35729
 2020-04-04 18:25:28.319  WARN  28064 --- [restartedMain] o.s.data.convert.CustomConversions       : Registering converter from class java.time.LocalDateTime to class java.time.Instant as reading converter although it doesn't convert from a store-supported type! You might wanna check you annotation setup at the converter implementation.
 2020-04-04 18:25:28.319  WARN  28064 --- [restartedMain] o.s.data.convert.CustomConversions       : Registering converter from class java.time.Instant to class java.time.LocalDateTime as reading converter although it doesn't convert from a store-supported type! You might wanna check you annotation setup at the converter implementation.
 2020-04-04 18:25:28.362  WARN  28064 --- [restartedMain] eactiveMongoPersistentEntityIndexCreator : Automatic index creation will be disabled by default as of Spring Data MongoDB 3.x.
	Please use 'MongoMappingContext#setAutoIndexCreation(boolean)' or override 'MongoConfigurationSupport#autoIndexCreation()' to be explicit.
	However, we recommend setting up indices manually in an application ready block. You may use index derivation there as well.

	> -----------------------------------------------------------------------------------------
	> @EventListener(ApplicationReadyEvent.class)
	> public void initIndicesAfterStartup() {
	>
	>     IndexOperations indexOps = mongoTemplate.indexOps(DomainType.class);
	>
	>     IndexResolver resolver = new MongoPersistentEntityIndexResolver(mongoMappingContext);
	>     resolver.resolveIndexFor(DomainType.class).forEach(indexOps::ensureIndex);
	> }
	> -----------------------------------------------------------------------------------------

 2020-04-04 18:25:28.425  INFO  28064 --- [restartedMain] org.mongodb.driver.cluster               : No server chosen by com.mongodb.async.client.ClientSessionHelper$1@4d1150fe from cluster description ClusterDescription{type=UNKNOWN, connectionMode=SINGLE, serverDescriptions=[ServerDescription{address=localhost:27017, type=UNKNOWN, state=CONNECTING, exception={com.mongodb.MongoSocketOpenException: Exception opening socket}, caused by {io.netty.channel.AbstractChannel$AnnotatedConnectException: Connection refused: localhost/0:0:0:0:0:0:0:1:27017}, caused by {java.net.ConnectException: Connection refused}}]}. Waiting for 30000 ms before timing out
 2020-04-04 18:25:28.438  INFO  28064 --- [restartedMain] org.mongodb.driver.cluster               : No server chosen by com.mongodb.async.client.ClientSessionHelper$1@4a881cf from cluster description ClusterDescription{type=UNKNOWN, connectionMode=SINGLE, serverDescriptions=[ServerDescription{address=localhost:27017, type=UNKNOWN, state=CONNECTING, exception={com.mongodb.MongoSocketOpenException: Exception opening socket}, caused by {io.netty.channel.AbstractChannel$AnnotatedConnectException: Connection refused: localhost/0:0:0:0:0:0:0:1:27017}, caused by {java.net.ConnectException: Connection refused}}]}. Waiting for 30000 ms before timing out
 2020-04-04 18:25:28.439  INFO  28064 --- [restartedMain] org.mongodb.driver.cluster               : No server chosen by com.mongodb.async.client.ClientSessionHelper$1@4fc10ec0 from cluster description ClusterDescription{type=UNKNOWN, connectionMode=SINGLE, serverDescriptions=[ServerDescription{address=localhost:27017, type=UNKNOWN, state=CONNECTING, exception={com.mongodb.MongoSocketOpenException: Exception opening socket}, caused by {io.netty.channel.AbstractChannel$AnnotatedConnectException: Connection refused: localhost/0:0:0:0:0:0:0:1:27017}, caused by {java.net.ConnectException: Connection refused}}]}. Waiting for 30000 ms before timing out
 2020-04-04 18:25:28.440  INFO  28064 --- [restartedMain] org.mongodb.driver.cluster               : No server chosen by com.mongodb.async.client.ClientSessionHelper$1@e11711a from cluster description ClusterDescription{type=UNKNOWN, connectionMode=SINGLE, serverDescriptions=[ServerDescription{address=localhost:27017, type=UNKNOWN, state=CONNECTING, exception={com.mongodb.MongoSocketOpenException: Exception opening socket}, caused by {io.netty.channel.AbstractChannel$AnnotatedConnectException: Connection refused: localhost/0:0:0:0:0:0:0:1:27017}, caused by {java.net.ConnectException: Connection refused}}]}. Waiting for 30000 ms before timing out
 2020-04-04 18:25:28.440  INFO  28064 --- [restartedMain] org.mongodb.driver.cluster               : No server chosen by com.mongodb.async.client.ClientSessionHelper$1@20f4b670 from cluster description ClusterDescription{type=UNKNOWN, connectionMode=SINGLE, serverDescriptions=[ServerDescription{address=localhost:27017, type=UNKNOWN, state=CONNECTING, exception={com.mongodb.MongoSocketOpenException: Exception opening socket}, caused by {io.netty.channel.AbstractChannel$AnnotatedConnectException: Connection refused: localhost/0:0:0:0:0:0:0:1:27017}, caused by {java.net.ConnectException: Connection refused}}]}. Waiting for 30000 ms before timing out
 2020-04-04 18:25:28.441  INFO  28064 --- [restartedMain] org.mongodb.driver.cluster               : No server chosen by com.mongodb.async.client.ClientSessionHelper$1@6012fd58 from cluster description ClusterDescription{type=UNKNOWN, connectionMode=SINGLE, serverDescriptions=[ServerDescription{address=localhost:27017, type=UNKNOWN, state=CONNECTING, exception={com.mongodb.MongoSocketOpenException: Exception opening socket}, caused by {io.netty.channel.AbstractChannel$AnnotatedConnectException: Connection refused: localhost/0:0:0:0:0:0:0:1:27017}, caused by {java.net.ConnectException: Connection refused}}]}. Waiting for 30000 ms before timing out
 2020-04-04 18:25:28.442  INFO  28064 --- [restartedMain] org.mongodb.driver.cluster               : No server chosen by com.mongodb.async.client.ClientSessionHelper$1@27e25894 from cluster description ClusterDescription{type=UNKNOWN, connectionMode=SINGLE, serverDescriptions=[ServerDescription{address=localhost:27017, type=UNKNOWN, state=CONNECTING, exception={com.mongodb.MongoSocketOpenException: Exception opening socket}, caused by {io.netty.channel.AbstractChannel$AnnotatedConnectException: Connection refused: localhost/0:0:0:0:0:0:0:1:27017}, caused by {java.net.ConnectException: Connection refused}}]}. Waiting for 30000 ms before timing out
 2020-04-04 18:25:28.444  INFO  28064 --- [restartedMain] org.mongodb.driver.cluster               : No server chosen by com.mongodb.async.client.ClientSessionHelper$1@58c7e486 from cluster description ClusterDescription{type=UNKNOWN, connectionMode=SINGLE, serverDescriptions=[ServerDescription{address=localhost:27017, type=UNKNOWN, state=CONNECTING, exception={com.mongodb.MongoSocketOpenException: Exception opening socket}, caused by {io.netty.channel.AbstractChannel$AnnotatedConnectException: Connection refused: localhost/0:0:0:0:0:0:0:1:27017}, caused by {java.net.ConnectException: Connection refused}}]}. Waiting for 30000 ms before timing out
 2020-04-04 18:25:28.446  INFO  28064 --- [restartedMain] org.mongodb.driver.cluster               : No server chosen by com.mongodb.async.client.ClientSessionHelper$1@25f657d3 from cluster description ClusterDescription{type=UNKNOWN, connectionMode=SINGLE, serverDescriptions=[ServerDescription{address=localhost:27017, type=UNKNOWN, state=CONNECTING, exception={com.mongodb.MongoSocketOpenException: Exception opening socket}, caused by {io.netty.channel.AbstractChannel$AnnotatedConnectException: Connection refused: localhost/0:0:0:0:0:0:0:1:27017}, caused by {java.net.ConnectException: Connection refused}}]}. Waiting for 30000 ms before timing out
 2020-04-04 18:25:28.447  INFO  28064 --- [restartedMain] org.mongodb.driver.cluster               : No server chosen by com.mongodb.async.client.ClientSessionHelper$1@3472b7ab from cluster description ClusterDescription{type=UNKNOWN, connectionMode=SINGLE, serverDescriptions=[ServerDescription{address=localhost:27017, type=UNKNOWN, state=CONNECTING, exception={com.mongodb.MongoSocketOpenException: Exception opening socket}, caused by {io.netty.channel.AbstractChannel$AnnotatedConnectException: Connection refused: localhost/0:0:0:0:0:0:0:1:27017}, caused by {java.net.ConnectException: Connection refused}}]}. Waiting for 30000 ms before timing out
 2020-04-04 18:25:28.929  INFO  28064 --- [restartedMain] o.s.s.concurrent.ThreadPoolTaskExecutor  : Initializing ExecutorService 'applicationTaskExecutor'
 2020-04-04 18:25:29.484  WARN  28064 --- [restartedMain] .m.c.i.MongoPersistentEntityIndexCreator : Automatic index creation will be disabled by default as of Spring Data MongoDB 3.x.
	Please use 'MongoMappingContext#setAutoIndexCreation(boolean)' or override 'MongoConfigurationSupport#autoIndexCreation()' to be explicit.
	However, we recommend setting up indices manually in an application ready block. You may use index derivation there as well.

	> -----------------------------------------------------------------------------------------
	> @EventListener(ApplicationReadyEvent.class)
	> public void initIndicesAfterStartup() {
	>
	>     IndexOperations indexOps = mongoTemplate.indexOps(DomainType.class);
	>
	>     IndexResolver resolver = new MongoPersistentEntityIndexResolver(mongoMappingContext);
	>     resolver.resolveIndexFor(DomainType.class).forEach(indexOps::ensureIndex);
	> }
	> -----------------------------------------------------------------------------------------

 2020-04-04 18:25:29.496  INFO  28064 --- [restartedMain] org.mongodb.driver.cluster               : Cluster description not yet available. Waiting for 30000 ms before timing out
 2020-04-04 18:25:58.438  INFO  28064 --- [cluster-5e88d1173ea5cc3c08d2aec1] org.mongodb.driver.cluster               : No server chosen by WritableServerSelector from cluster description ClusterDescription{type=UNKNOWN, connectionMode=SINGLE, serverDescriptions=[ServerDescription{address=localhost:27017, type=UNKNOWN, state=CONNECTING, exception={com.mongodb.MongoSocketOpenException: Exception opening socket}, caused by {io.netty.channel.AbstractChannel$AnnotatedConnectException: Connection refused: localhost/0:0:0:0:0:0:0:1:27017}, caused by {java.net.ConnectException: Connection refused}}]}. Waiting for 30000 ms before timing out
 2020-04-04 18:25:58.438  INFO  28064 --- [cluster-5e88d1173ea5cc3c08d2aec1] org.mongodb.driver.cluster               : No server chosen by WritableServerSelector from cluster description ClusterDescription{type=UNKNOWN, connectionMode=SINGLE, serverDescriptions=[ServerDescription{address=localhost:27017, type=UNKNOWN, state=CONNECTING, exception={com.mongodb.MongoSocketOpenException: Exception opening socket}, caused by {io.netty.channel.AbstractChannel$AnnotatedConnectException: Connection refused: localhost/0:0:0:0:0:0:0:1:27017}, caused by {java.net.ConnectException: Connection refused}}]}. Waiting for 30000 ms before timing out
 2020-04-04 18:25:58.439  INFO  28064 --- [cluster-5e88d1173ea5cc3c08d2aec1] org.mongodb.driver.cluster               : No server chosen by WritableServerSelector from cluster description ClusterDescription{type=UNKNOWN, connectionMode=SINGLE, serverDescriptions=[ServerDescription{address=localhost:27017, type=UNKNOWN, state=CONNECTING, exception={com.mongodb.MongoSocketOpenException: Exception opening socket}, caused by {io.netty.channel.AbstractChannel$AnnotatedConnectException: Connection refused: localhost/0:0:0:0:0:0:0:1:27017}, caused by {java.net.ConnectException: Connection refused}}]}. Waiting for 30000 ms before timing out
 2020-04-04 18:25:58.440  INFO  28064 --- [cluster-5e88d1173ea5cc3c08d2aec1] org.mongodb.driver.cluster               : No server chosen by WritableServerSelector from cluster description ClusterDescription{type=UNKNOWN, connectionMode=SINGLE, serverDescriptions=[ServerDescription{address=localhost:27017, type=UNKNOWN, state=CONNECTING, exception={com.mongodb.MongoSocketOpenException: Exception opening socket}, caused by {io.netty.channel.AbstractChannel$AnnotatedConnectException: Connection refused: localhost/0:0:0:0:0:0:0:1:27017}, caused by {java.net.ConnectException: Connection refused}}]}. Waiting for 30000 ms before timing out
 2020-04-04 18:25:58.440  INFO  28064 --- [cluster-5e88d1173ea5cc3c08d2aec1] org.mongodb.driver.cluster               : No server chosen by WritableServerSelector from cluster description ClusterDescription{type=UNKNOWN, connectionMode=SINGLE, serverDescriptions=[ServerDescription{address=localhost:27017, type=UNKNOWN, state=CONNECTING, exception={com.mongodb.MongoSocketOpenException: Exception opening socket}, caused by {io.netty.channel.AbstractChannel$AnnotatedConnectException: Connection refused: localhost/0:0:0:0:0:0:0:1:27017}, caused by {java.net.ConnectException: Connection refused}}]}. Waiting for 30000 ms before timing out
 2020-04-04 18:25:58.441  INFO  28064 --- [cluster-5e88d1173ea5cc3c08d2aec1] org.mongodb.driver.cluster               : No server chosen by WritableServerSelector from cluster description ClusterDescription{type=UNKNOWN, connectionMode=SINGLE, serverDescriptions=[ServerDescription{address=localhost:27017, type=UNKNOWN, state=CONNECTING, exception={com.mongodb.MongoSocketOpenException: Exception opening socket}, caused by {io.netty.channel.AbstractChannel$AnnotatedConnectException: Connection refused: localhost/0:0:0:0:0:0:0:1:27017}, caused by {java.net.ConnectException: Connection refused}}]}. Waiting for 30000 ms before timing out
 2020-04-04 18:25:58.441  INFO  28064 --- [cluster-5e88d1173ea5cc3c08d2aec1] org.mongodb.driver.cluster               : No server chosen by WritableServerSelector from cluster description ClusterDescription{type=UNKNOWN, connectionMode=SINGLE, serverDescriptions=[ServerDescription{address=localhost:27017, type=UNKNOWN, state=CONNECTING, exception={com.mongodb.MongoSocketOpenException: Exception opening socket}, caused by {io.netty.channel.AbstractChannel$AnnotatedConnectException: Connection refused: localhost/0:0:0:0:0:0:0:1:27017}, caused by {java.net.ConnectException: Connection refused}}]}. Waiting for 30000 ms before timing out
 2020-04-04 18:25:58.444  INFO  28064 --- [cluster-5e88d1173ea5cc3c08d2aec1] org.mongodb.driver.cluster               : No server chosen by WritableServerSelector from cluster description ClusterDescription{type=UNKNOWN, connectionMode=SINGLE, serverDescriptions=[ServerDescription{address=localhost:27017, type=UNKNOWN, state=CONNECTING, exception={com.mongodb.MongoSocketOpenException: Exception opening socket}, caused by {io.netty.channel.AbstractChannel$AnnotatedConnectException: Connection refused: localhost/0:0:0:0:0:0:0:1:27017}, caused by {java.net.ConnectException: Connection refused}}]}. Waiting for 30000 ms before timing out
 2020-04-04 18:25:58.445  INFO  28064 --- [cluster-5e88d1173ea5cc3c08d2aec1] org.mongodb.driver.cluster               : No server chosen by WritableServerSelector from cluster description ClusterDescription{type=UNKNOWN, connectionMode=SINGLE, serverDescriptions=[ServerDescription{address=localhost:27017, type=UNKNOWN, state=CONNECTING, exception={com.mongodb.MongoSocketOpenException: Exception opening socket}, caused by {io.netty.channel.AbstractChannel$AnnotatedConnectException: Connection refused: localhost/0:0:0:0:0:0:0:1:27017}, caused by {java.net.ConnectException: Connection refused}}]}. Waiting for 30000 ms before timing out
 2020-04-04 18:25:58.447  INFO  28064 --- [cluster-5e88d1173ea5cc3c08d2aec1] org.mongodb.driver.cluster               : No server chosen by WritableServerSelector from cluster description ClusterDescription{type=UNKNOWN, connectionMode=SINGLE, serverDescriptions=[ServerDescription{address=localhost:27017, type=UNKNOWN, state=CONNECTING, exception={com.mongodb.MongoSocketOpenException: Exception opening socket}, caused by {io.netty.channel.AbstractChannel$AnnotatedConnectException: Connection refused: localhost/0:0:0:0:0:0:0:1:27017}, caused by {java.net.ConnectException: Connection refused}}]}. Waiting for 30000 ms before timing out
 2020-04-04 18:25:59.504  WARN  28064 --- [restartedMain] ConfigServletWebServerApplicationContext : Exception encountered during context initialization - cancelling refresh attempt: org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'mongoTemplate' defined in class path resource [org/springframework/boot/autoconfigure/data/mongo/MongoDbFactoryDependentConfiguration.class]: Bean instantiation via factory method failed; nested exception is org.springframework.beans.BeanInstantiationException: Failed to instantiate [org.springframework.data.mongodb.core.MongoTemplate]: Factory method 'mongoTemplate' threw exception; nested exception is org.springframework.dao.DataAccessResourceFailureException: Timed out after 30000 ms while waiting to connect. Client view of cluster state is {type=UNKNOWN, servers=[{address=localhost:27017, type=UNKNOWN, state=CONNECTING, exception={com.mongodb.MongoSocketOpenException: Exception opening socket}, caused by {java.net.ConnectException: Connection refused (Connection refused)}}]; nested exception is com.mongodb.MongoTimeoutException: Timed out after 30000 ms while waiting to connect. Client view of cluster state is {type=UNKNOWN, servers=[{address=localhost:27017, type=UNKNOWN, state=CONNECTING, exception={com.mongodb.MongoSocketOpenException: Exception opening socket}, caused by {java.net.ConnectException: Connection refused (Connection refused)}}]
 2020-04-04 18:25:59.505  INFO  28064 --- [restartedMain] o.s.s.concurrent.ThreadPoolTaskExecutor  : Shutting down ExecutorService 'applicationTaskExecutor'
 2020-04-04 18:25:59.529 ERROR  28064 --- [cluster-5e88d1173ea5cc3c08d2aec1] o.s.d.m.core.ReactiveMongoTemplate       : Unexpected exception during asynchronous execution

org.springframework.data.mongodb.UncategorizedMongoDbException: Shutdown in progress; nested exception is com.mongodb.MongoClientException: Shutdown in progress
	at org.springframework.data.mongodb.core.MongoExceptionTranslator.translateExceptionIfPossible(MongoExceptionTranslator.java:138) ~[spring-data-mongodb-2.2.0.RELEASE.jar:2.2.0.RELEASE]
	at org.springframework.data.mongodb.core.ReactiveMongoTemplate.potentiallyConvertRuntimeException(ReactiveMongoTemplate.java:2777) ~[spring-data-mongodb-2.2.0.RELEASE.jar:2.2.0.RELEASE]
	at org.springframework.data.mongodb.core.ReactiveMongoTemplate.lambda$translateException$87(ReactiveMongoTemplate.java:2760) ~[spring-data-mongodb-2.2.0.RELEASE.jar:2.2.0.RELEASE]
	at reactor.core.publisher.Flux.lambda$onErrorMap$28(Flux.java:6390) ~[reactor-core-3.3.0.RELEASE.jar:3.3.0.RELEASE]
	at reactor.core.publisher.FluxOnErrorResume$ResumeSubscriber.onError(FluxOnErrorResume.java:88) ~[reactor-core-3.3.0.RELEASE.jar:3.3.0.RELEASE]
	at reactor.core.publisher.MonoFlatMapMany$FlatMapManyInner.onError(MonoFlatMapMany.java:247) ~[reactor-core-3.3.0.RELEASE.jar:3.3.0.RELEASE]
	at com.mongodb.reactivestreams.client.internal.SingleResultObservableToPublisher$1.onError(SingleResultObservableToPublisher.java:50) ~[mongodb-driver-reactivestreams-1.12.0.jar:na]
	at com.mongodb.reactivestreams.client.internal.ObservableToPublisher$1.onError(ObservableToPublisher.java:71) ~[mongodb-driver-reactivestreams-1.12.0.jar:na]
	at com.mongodb.async.client.AbstractSubscription.onError(AbstractSubscription.java:129) ~[mongodb-driver-async-3.11.1.jar:na]
	at com.mongodb.async.client.SingleResultCallbackSubscription$1.onResult(SingleResultCallbackSubscription.java:42) ~[mongodb-driver-async-3.11.1.jar:na]
	at com.mongodb.async.client.MongoCollectionImpl$7.onResult(MongoCollectionImpl.java:921) ~[mongodb-driver-async-3.11.1.jar:na]
	at com.mongodb.async.client.MongoCollectionImpl$7.onResult(MongoCollectionImpl.java:917) ~[mongodb-driver-async-3.11.1.jar:na]
	at com.mongodb.async.client.MongoCollectionImpl$9.onResult(MongoCollectionImpl.java:980) ~[mongodb-driver-async-3.11.1.jar:na]
	at com.mongodb.async.client.MongoCollectionImpl$9.onResult(MongoCollectionImpl.java:976) ~[mongodb-driver-async-3.11.1.jar:na]
	at com.mongodb.internal.async.ErrorHandlingResultCallback.onResult(ErrorHandlingResultCallback.java:49) ~[mongodb-driver-core-3.11.1.jar:na]
	at com.mongodb.async.client.OperationExecutorImpl$2$1$1.onResult(OperationExecutorImpl.java:140) ~[mongodb-driver-async-3.11.1.jar:na]
	at com.mongodb.internal.async.ErrorHandlingResultCallback.onResult(ErrorHandlingResultCallback.java:49) ~[mongodb-driver-core-3.11.1.jar:na]
	at com.mongodb.operation.CreateIndexesOperation$2.call(CreateIndexesOperation.java:194) ~[mongodb-driver-core-3.11.1.jar:na]
	at com.mongodb.operation.OperationHelper$AsyncCallableWithConnectionCallback.onResult(OperationHelper.java:569) ~[mongodb-driver-core-3.11.1.jar:na]
	at com.mongodb.operation.OperationHelper$AsyncCallableWithConnectionCallback.onResult(OperationHelper.java:561) ~[mongodb-driver-core-3.11.1.jar:na]
	at com.mongodb.internal.async.ErrorHandlingResultCallback.onResult(ErrorHandlingResultCallback.java:49) ~[mongodb-driver-core-3.11.1.jar:na]
	at com.mongodb.binding.AsyncClusterBinding$1.onResult(AsyncClusterBinding.java:111) ~[mongodb-driver-core-3.11.1.jar:na]
	at com.mongodb.binding.AsyncClusterBinding$1.onResult(AsyncClusterBinding.java:107) ~[mongodb-driver-core-3.11.1.jar:na]
	at com.mongodb.internal.connection.BaseCluster$ServerSelectionRequest.onResult(BaseCluster.java:440) ~[mongodb-driver-core-3.11.1.jar:na]
	at com.mongodb.internal.connection.BaseCluster$WaitQueueHandler.run(BaseCluster.java:512) ~[mongodb-driver-core-3.11.1.jar:na]
	at java.base/java.lang.Thread.run(Thread.java:844) ~[na:na]
Caused by: com.mongodb.MongoClientException: Shutdown in progress
	... 2 common frames omitted

 2020-04-04 18:25:59.542 ERROR  28064 --- [cluster-5e88d1173ea5cc3c08d2aec1] o.s.d.m.core.ReactiveMongoTemplate       : Unexpected exception during asynchronous execution

org.springframework.data.mongodb.UncategorizedMongoDbException: Shutdown in progress; nested exception is com.mongodb.MongoClientException: Shutdown in progress
	at org.springframework.data.mongodb.core.MongoExceptionTranslator.translateExceptionIfPossible(MongoExceptionTranslator.java:138) ~[spring-data-mongodb-2.2.0.RELEASE.jar:2.2.0.RELEASE]
	at org.springframework.data.mongodb.core.ReactiveMongoTemplate.potentiallyConvertRuntimeException(ReactiveMongoTemplate.java:2777) ~[spring-data-mongodb-2.2.0.RELEASE.jar:2.2.0.RELEASE]
	at org.springframework.data.mongodb.core.ReactiveMongoTemplate.lambda$translateException$87(ReactiveMongoTemplate.java:2760) ~[spring-data-mongodb-2.2.0.RELEASE.jar:2.2.0.RELEASE]
	at reactor.core.publisher.Flux.lambda$onErrorMap$28(Flux.java:6390) ~[reactor-core-3.3.0.RELEASE.jar:3.3.0.RELEASE]
	at reactor.core.publisher.FluxOnErrorResume$ResumeSubscriber.onError(FluxOnErrorResume.java:88) ~[reactor-core-3.3.0.RELEASE.jar:3.3.0.RELEASE]
	at reactor.core.publisher.MonoFlatMapMany$FlatMapManyInner.onError(MonoFlatMapMany.java:247) ~[reactor-core-3.3.0.RELEASE.jar:3.3.0.RELEASE]
	at com.mongodb.reactivestreams.client.internal.SingleResultObservableToPublisher$1.onError(SingleResultObservableToPublisher.java:50) ~[mongodb-driver-reactivestreams-1.12.0.jar:na]
	at com.mongodb.reactivestreams.client.internal.ObservableToPublisher$1.onError(ObservableToPublisher.java:71) ~[mongodb-driver-reactivestreams-1.12.0.jar:na]
	at com.mongodb.async.client.AbstractSubscription.onError(AbstractSubscription.java:129) ~[mongodb-driver-async-3.11.1.jar:na]
	at com.mongodb.async.client.SingleResultCallbackSubscription$1.onResult(SingleResultCallbackSubscription.java:42) ~[mongodb-driver-async-3.11.1.jar:na]
	at com.mongodb.async.client.MongoCollectionImpl$7.onResult(MongoCollectionImpl.java:921) ~[mongodb-driver-async-3.11.1.jar:na]
	at com.mongodb.async.client.MongoCollectionImpl$7.onResult(MongoCollectionImpl.java:917) ~[mongodb-driver-async-3.11.1.jar:na]
	at com.mongodb.async.client.MongoCollectionImpl$9.onResult(MongoCollectionImpl.java:980) ~[mongodb-driver-async-3.11.1.jar:na]
	at com.mongodb.async.client.MongoCollectionImpl$9.onResult(MongoCollectionImpl.java:976) ~[mongodb-driver-async-3.11.1.jar:na]
	at com.mongodb.internal.async.ErrorHandlingResultCallback.onResult(ErrorHandlingResultCallback.java:49) ~[mongodb-driver-core-3.11.1.jar:na]
	at com.mongodb.async.client.OperationExecutorImpl$2$1$1.onResult(OperationExecutorImpl.java:140) ~[mongodb-driver-async-3.11.1.jar:na]
	at com.mongodb.internal.async.ErrorHandlingResultCallback.onResult(ErrorHandlingResultCallback.java:49) ~[mongodb-driver-core-3.11.1.jar:na]
	at com.mongodb.operation.CreateIndexesOperation$2.call(CreateIndexesOperation.java:194) ~[mongodb-driver-core-3.11.1.jar:na]
	at com.mongodb.operation.OperationHelper$AsyncCallableWithConnectionCallback.onResult(OperationHelper.java:569) ~[mongodb-driver-core-3.11.1.jar:na]
	at com.mongodb.operation.OperationHelper$AsyncCallableWithConnectionCallback.onResult(OperationHelper.java:561) ~[mongodb-driver-core-3.11.1.jar:na]
	at com.mongodb.internal.async.ErrorHandlingResultCallback.onResult(ErrorHandlingResultCallback.java:49) ~[mongodb-driver-core-3.11.1.jar:na]
	at com.mongodb.binding.AsyncClusterBinding$1.onResult(AsyncClusterBinding.java:111) ~[mongodb-driver-core-3.11.1.jar:na]
	at com.mongodb.binding.AsyncClusterBinding$1.onResult(AsyncClusterBinding.java:107) ~[mongodb-driver-core-3.11.1.jar:na]
	at com.mongodb.internal.connection.BaseCluster$ServerSelectionRequest.onResult(BaseCluster.java:440) ~[mongodb-driver-core-3.11.1.jar:na]
	at com.mongodb.internal.connection.BaseCluster$WaitQueueHandler.run(BaseCluster.java:512) ~[mongodb-driver-core-3.11.1.jar:na]
	at java.base/java.lang.Thread.run(Thread.java:844) ~[na:na]
Caused by: com.mongodb.MongoClientException: Shutdown in progress
	... 2 common frames omitted

 2020-04-04 18:25:59.543 ERROR  28064 --- [cluster-5e88d1173ea5cc3c08d2aec1] org.mongodb.driver.client                : Callback onResult call produced an error

com.mongodb.MongoClientException: Shutdown in progress
	at com.mongodb.internal.connection.BaseCluster$WaitQueueHandler.run(BaseCluster.java:512) [mongodb-driver-core-3.11.1.jar:na]
	at java.base/java.lang.Thread.run(Thread.java:844) [na:na]

 2020-04-04 18:25:59.550 ERROR  28064 --- [cluster-5e88d1173ea5cc3c08d2aec1] o.s.d.m.core.ReactiveMongoTemplate       : Unexpected exception during asynchronous execution

org.springframework.data.mongodb.UncategorizedMongoDbException: Shutdown in progress; nested exception is com.mongodb.MongoClientException: Shutdown in progress
	at org.springframework.data.mongodb.core.MongoExceptionTranslator.translateExceptionIfPossible(MongoExceptionTranslator.java:138) ~[spring-data-mongodb-2.2.0.RELEASE.jar:2.2.0.RELEASE]
	at org.springframework.data.mongodb.core.ReactiveMongoTemplate.potentiallyConvertRuntimeException(ReactiveMongoTemplate.java:2777) ~[spring-data-mongodb-2.2.0.RELEASE.jar:2.2.0.RELEASE]
	at org.springframework.data.mongodb.core.ReactiveMongoTemplate.lambda$translateException$87(ReactiveMongoTemplate.java:2760) ~[spring-data-mongodb-2.2.0.RELEASE.jar:2.2.0.RELEASE]
	at reactor.core.publisher.Flux.lambda$onErrorMap$28(Flux.java:6390) ~[reactor-core-3.3.0.RELEASE.jar:3.3.0.RELEASE]
	at reactor.core.publisher.FluxOnErrorResume$ResumeSubscriber.onError(FluxOnErrorResume.java:88) ~[reactor-core-3.3.0.RELEASE.jar:3.3.0.RELEASE]
	at reactor.core.publisher.MonoFlatMapMany$FlatMapManyInner.onError(MonoFlatMapMany.java:247) ~[reactor-core-3.3.0.RELEASE.jar:3.3.0.RELEASE]
	at com.mongodb.reactivestreams.client.internal.SingleResultObservableToPublisher$1.onError(SingleResultObservableToPublisher.java:50) ~[mongodb-driver-reactivestreams-1.12.0.jar:na]
	at com.mongodb.reactivestreams.client.internal.ObservableToPublisher$1.onError(ObservableToPublisher.java:71) ~[mongodb-driver-reactivestreams-1.12.0.jar:na]
	at com.mongodb.async.client.AbstractSubscription.onError(AbstractSubscription.java:129) ~[mongodb-driver-async-3.11.1.jar:na]
	at com.mongodb.async.client.SingleResultCallbackSubscription$1.onResult(SingleResultCallbackSubscription.java:42) ~[mongodb-driver-async-3.11.1.jar:na]
	at com.mongodb.async.client.MongoCollectionImpl$7.onResult(MongoCollectionImpl.java:921) ~[mongodb-driver-async-3.11.1.jar:na]
	at com.mongodb.async.client.MongoCollectionImpl$7.onResult(MongoCollectionImpl.java:917) ~[mongodb-driver-async-3.11.1.jar:na]
	at com.mongodb.async.client.MongoCollectionImpl$9.onResult(MongoCollectionImpl.java:980) ~[mongodb-driver-async-3.11.1.jar:na]
	at com.mongodb.async.client.MongoCollectionImpl$9.onResult(MongoCollectionImpl.java:976) ~[mongodb-driver-async-3.11.1.jar:na]
	at com.mongodb.internal.async.ErrorHandlingResultCallback.onResult(ErrorHandlingResultCallback.java:49) ~[mongodb-driver-core-3.11.1.jar:na]
	at com.mongodb.async.client.OperationExecutorImpl$2$1$1.onResult(OperationExecutorImpl.java:140) ~[mongodb-driver-async-3.11.1.jar:na]
	at com.mongodb.internal.async.ErrorHandlingResultCallback.onResult(ErrorHandlingResultCallback.java:49) ~[mongodb-driver-core-3.11.1.jar:na]
	at com.mongodb.operation.CreateIndexesOperation$2.call(CreateIndexesOperation.java:194) ~[mongodb-driver-core-3.11.1.jar:na]
	at com.mongodb.operation.OperationHelper$AsyncCallableWithConnectionCallback.onResult(OperationHelper.java:569) ~[mongodb-driver-core-3.11.1.jar:na]
	at com.mongodb.operation.OperationHelper$AsyncCallableWithConnectionCallback.onResult(OperationHelper.java:561) ~[mongodb-driver-core-3.11.1.jar:na]
	at com.mongodb.internal.async.ErrorHandlingResultCallback.onResult(ErrorHandlingResultCallback.java:49) ~[mongodb-driver-core-3.11.1.jar:na]
	at com.mongodb.binding.AsyncClusterBinding$1.onResult(AsyncClusterBinding.java:111) ~[mongodb-driver-core-3.11.1.jar:na]
	at com.mongodb.binding.AsyncClusterBinding$1.onResult(AsyncClusterBinding.java:107) ~[mongodb-driver-core-3.11.1.jar:na]
	at com.mongodb.internal.connection.BaseCluster$ServerSelectionRequest.onResult(BaseCluster.java:440) ~[mongodb-driver-core-3.11.1.jar:na]
	at com.mongodb.internal.connection.BaseCluster$WaitQueueHandler.run(BaseCluster.java:512) ~[mongodb-driver-core-3.11.1.jar:na]
	at java.base/java.lang.Thread.run(Thread.java:844) ~[na:na]
Caused by: com.mongodb.MongoClientException: Shutdown in progress
	... 2 common frames omitted

 2020-04-04 18:25:59.551 ERROR  28064 --- [cluster-5e88d1173ea5cc3c08d2aec1] org.mongodb.driver.client                : Callback onResult call produced an error

com.mongodb.MongoClientException: Shutdown in progress
	at com.mongodb.internal.connection.BaseCluster$WaitQueueHandler.run(BaseCluster.java:512) [mongodb-driver-core-3.11.1.jar:na]
	at java.base/java.lang.Thread.run(Thread.java:844) [na:na]

 2020-04-04 18:25:59.557 ERROR  28064 --- [cluster-5e88d1173ea5cc3c08d2aec1] o.s.d.m.core.ReactiveMongoTemplate       : Unexpected exception during asynchronous execution

org.springframework.data.mongodb.UncategorizedMongoDbException: Shutdown in progress; nested exception is com.mongodb.MongoClientException: Shutdown in progress
	at org.springframework.data.mongodb.core.MongoExceptionTranslator.translateExceptionIfPossible(MongoExceptionTranslator.java:138) ~[spring-data-mongodb-2.2.0.RELEASE.jar:2.2.0.RELEASE]
	at org.springframework.data.mongodb.core.ReactiveMongoTemplate.potentiallyConvertRuntimeException(ReactiveMongoTemplate.java:2777) ~[spring-data-mongodb-2.2.0.RELEASE.jar:2.2.0.RELEASE]
	at org.springframework.data.mongodb.core.ReactiveMongoTemplate.lambda$translateException$87(ReactiveMongoTemplate.java:2760) ~[spring-data-mongodb-2.2.0.RELEASE.jar:2.2.0.RELEASE]
	at reactor.core.publisher.Flux.lambda$onErrorMap$28(Flux.java:6390) ~[reactor-core-3.3.0.RELEASE.jar:3.3.0.RELEASE]
	at reactor.core.publisher.FluxOnErrorResume$ResumeSubscriber.onError(FluxOnErrorResume.java:88) ~[reactor-core-3.3.0.RELEASE.jar:3.3.0.RELEASE]
	at reactor.core.publisher.MonoFlatMapMany$FlatMapManyInner.onError(MonoFlatMapMany.java:247) ~[reactor-core-3.3.0.RELEASE.jar:3.3.0.RELEASE]
	at com.mongodb.reactivestreams.client.internal.SingleResultObservableToPublisher$1.onError(SingleResultObservableToPublisher.java:50) ~[mongodb-driver-reactivestreams-1.12.0.jar:na]
	at com.mongodb.reactivestreams.client.internal.ObservableToPublisher$1.onError(ObservableToPublisher.java:71) ~[mongodb-driver-reactivestreams-1.12.0.jar:na]
	at com.mongodb.async.client.AbstractSubscription.onError(AbstractSubscription.java:129) ~[mongodb-driver-async-3.11.1.jar:na]
	at com.mongodb.async.client.SingleResultCallbackSubscription$1.onResult(SingleResultCallbackSubscription.java:42) ~[mongodb-driver-async-3.11.1.jar:na]
	at com.mongodb.async.client.MongoCollectionImpl$7.onResult(MongoCollectionImpl.java:921) ~[mongodb-driver-async-3.11.1.jar:na]
	at com.mongodb.async.client.MongoCollectionImpl$7.onResult(MongoCollectionImpl.java:917) ~[mongodb-driver-async-3.11.1.jar:na]
	at com.mongodb.async.client.MongoCollectionImpl$9.onResult(MongoCollectionImpl.java:980) ~[mongodb-driver-async-3.11.1.jar:na]
	at com.mongodb.async.client.MongoCollectionImpl$9.onResult(MongoCollectionImpl.java:976) ~[mongodb-driver-async-3.11.1.jar:na]
	at com.mongodb.internal.async.ErrorHandlingResultCallback.onResult(ErrorHandlingResultCallback.java:49) ~[mongodb-driver-core-3.11.1.jar:na]
	at com.mongodb.async.client.OperationExecutorImpl$2$1$1.onResult(OperationExecutorImpl.java:140) ~[mongodb-driver-async-3.11.1.jar:na]
	at com.mongodb.internal.async.ErrorHandlingResultCallback.onResult(ErrorHandlingResultCallback.java:49) ~[mongodb-driver-core-3.11.1.jar:na]
	at com.mongodb.operation.CreateIndexesOperation$2.call(CreateIndexesOperation.java:194) ~[mongodb-driver-core-3.11.1.jar:na]
	at com.mongodb.operation.OperationHelper$AsyncCallableWithConnectionCallback.onResult(OperationHelper.java:569) ~[mongodb-driver-core-3.11.1.jar:na]
	at com.mongodb.operation.OperationHelper$AsyncCallableWithConnectionCallback.onResult(OperationHelper.java:561) ~[mongodb-driver-core-3.11.1.jar:na]
	at com.mongodb.internal.async.ErrorHandlingResultCallback.onResult(ErrorHandlingResultCallback.java:49) ~[mongodb-driver-core-3.11.1.jar:na]
	at com.mongodb.binding.AsyncClusterBinding$1.onResult(AsyncClusterBinding.java:111) ~[mongodb-driver-core-3.11.1.jar:na]
	at com.mongodb.binding.AsyncClusterBinding$1.onResult(AsyncClusterBinding.java:107) ~[mongodb-driver-core-3.11.1.jar:na]
	at com.mongodb.internal.connection.BaseCluster$ServerSelectionRequest.onResult(BaseCluster.java:440) ~[mongodb-driver-core-3.11.1.jar:na]
	at com.mongodb.internal.connection.BaseCluster$WaitQueueHandler.run(BaseCluster.java:512) ~[mongodb-driver-core-3.11.1.jar:na]
	at java.base/java.lang.Thread.run(Thread.java:844) ~[na:na]
Caused by: com.mongodb.MongoClientException: Shutdown in progress
	... 2 common frames omitted

 2020-04-04 18:25:59.559 ERROR  28064 --- [cluster-5e88d1173ea5cc3c08d2aec1] org.mongodb.driver.client                : Callback onResult call produced an error

com.mongodb.MongoClientException: Shutdown in progress
	at com.mongodb.internal.connection.BaseCluster$WaitQueueHandler.run(BaseCluster.java:512) [mongodb-driver-core-3.11.1.jar:na]
	at java.base/java.lang.Thread.run(Thread.java:844) [na:na]

 2020-04-04 18:25:59.563 ERROR  28064 --- [cluster-5e88d1173ea5cc3c08d2aec1] o.s.d.m.core.ReactiveMongoTemplate       : Unexpected exception during asynchronous execution

org.springframework.data.mongodb.UncategorizedMongoDbException: Shutdown in progress; nested exception is com.mongodb.MongoClientException: Shutdown in progress
	at org.springframework.data.mongodb.core.MongoExceptionTranslator.translateExceptionIfPossible(MongoExceptionTranslator.java:138) ~[spring-data-mongodb-2.2.0.RELEASE.jar:2.2.0.RELEASE]
	at org.springframework.data.mongodb.core.ReactiveMongoTemplate.potentiallyConvertRuntimeException(ReactiveMongoTemplate.java:2777) ~[spring-data-mongodb-2.2.0.RELEASE.jar:2.2.0.RELEASE]
	at org.springframework.data.mongodb.core.ReactiveMongoTemplate.lambda$translateException$87(ReactiveMongoTemplate.java:2760) ~[spring-data-mongodb-2.2.0.RELEASE.jar:2.2.0.RELEASE]
	at reactor.core.publisher.Flux.lambda$onErrorMap$28(Flux.java:6390) ~[reactor-core-3.3.0.RELEASE.jar:3.3.0.RELEASE]
	at reactor.core.publisher.FluxOnErrorResume$ResumeSubscriber.onError(FluxOnErrorResume.java:88) ~[reactor-core-3.3.0.RELEASE.jar:3.3.0.RELEASE]
	at reactor.core.publisher.MonoFlatMapMany$FlatMapManyInner.onError(MonoFlatMapMany.java:247) ~[reactor-core-3.3.0.RELEASE.jar:3.3.0.RELEASE]
	at com.mongodb.reactivestreams.client.internal.SingleResultObservableToPublisher$1.onError(SingleResultObservableToPublisher.java:50) ~[mongodb-driver-reactivestreams-1.12.0.jar:na]
	at com.mongodb.reactivestreams.client.internal.ObservableToPublisher$1.onError(ObservableToPublisher.java:71) ~[mongodb-driver-reactivestreams-1.12.0.jar:na]
	at com.mongodb.async.client.AbstractSubscription.onError(AbstractSubscription.java:129) ~[mongodb-driver-async-3.11.1.jar:na]
	at com.mongodb.async.client.SingleResultCallbackSubscription$1.onResult(SingleResultCallbackSubscription.java:42) ~[mongodb-driver-async-3.11.1.jar:na]
	at com.mongodb.async.client.MongoCollectionImpl$7.onResult(MongoCollectionImpl.java:921) ~[mongodb-driver-async-3.11.1.jar:na]
	at com.mongodb.async.client.MongoCollectionImpl$7.onResult(MongoCollectionImpl.java:917) ~[mongodb-driver-async-3.11.1.jar:na]
	at com.mongodb.async.client.MongoCollectionImpl$9.onResult(MongoCollectionImpl.java:980) ~[mongodb-driver-async-3.11.1.jar:na]
	at com.mongodb.async.client.MongoCollectionImpl$9.onResult(MongoCollectionImpl.java:976) ~[mongodb-driver-async-3.11.1.jar:na]
	at com.mongodb.internal.async.ErrorHandlingResultCallback.onResult(ErrorHandlingResultCallback.java:49) ~[mongodb-driver-core-3.11.1.jar:na]
	at com.mongodb.async.client.OperationExecutorImpl$2$1$1.onResult(OperationExecutorImpl.java:140) ~[mongodb-driver-async-3.11.1.jar:na]
	at com.mongodb.internal.async.ErrorHandlingResultCallback.onResult(ErrorHandlingResultCallback.java:49) ~[mongodb-driver-core-3.11.1.jar:na]
	at com.mongodb.operation.CreateIndexesOperation$2.call(CreateIndexesOperation.java:194) ~[mongodb-driver-core-3.11.1.jar:na]
	at com.mongodb.operation.OperationHelper$AsyncCallableWithConnectionCallback.onResult(OperationHelper.java:569) ~[mongodb-driver-core-3.11.1.jar:na]
	at com.mongodb.operation.OperationHelper$AsyncCallableWithConnectionCallback.onResult(OperationHelper.java:561) ~[mongodb-driver-core-3.11.1.jar:na]
	at com.mongodb.internal.async.ErrorHandlingResultCallback.onResult(ErrorHandlingResultCallback.java:49) ~[mongodb-driver-core-3.11.1.jar:na]
	at com.mongodb.binding.AsyncClusterBinding$1.onResult(AsyncClusterBinding.java:111) ~[mongodb-driver-core-3.11.1.jar:na]
	at com.mongodb.binding.AsyncClusterBinding$1.onResult(AsyncClusterBinding.java:107) ~[mongodb-driver-core-3.11.1.jar:na]
	at com.mongodb.internal.connection.BaseCluster$ServerSelectionRequest.onResult(BaseCluster.java:440) ~[mongodb-driver-core-3.11.1.jar:na]
	at com.mongodb.internal.connection.BaseCluster$WaitQueueHandler.run(BaseCluster.java:512) ~[mongodb-driver-core-3.11.1.jar:na]
	at java.base/java.lang.Thread.run(Thread.java:844) ~[na:na]
Caused by: com.mongodb.MongoClientException: Shutdown in progress
	... 2 common frames omitted

 2020-04-04 18:25:59.566 ERROR  28064 --- [cluster-5e88d1173ea5cc3c08d2aec1] o.s.d.m.core.ReactiveMongoTemplate       : Unexpected exception during asynchronous execution

org.springframework.data.mongodb.UncategorizedMongoDbException: Shutdown in progress; nested exception is com.mongodb.MongoClientException: Shutdown in progress
	at org.springframework.data.mongodb.core.MongoExceptionTranslator.translateExceptionIfPossible(MongoExceptionTranslator.java:138) ~[spring-data-mongodb-2.2.0.RELEASE.jar:2.2.0.RELEASE]
	at org.springframework.data.mongodb.core.ReactiveMongoTemplate.potentiallyConvertRuntimeException(ReactiveMongoTemplate.java:2777) ~[spring-data-mongodb-2.2.0.RELEASE.jar:2.2.0.RELEASE]
	at org.springframework.data.mongodb.core.ReactiveMongoTemplate.lambda$translateException$87(ReactiveMongoTemplate.java:2760) ~[spring-data-mongodb-2.2.0.RELEASE.jar:2.2.0.RELEASE]
	at reactor.core.publisher.Flux.lambda$onErrorMap$28(Flux.java:6390) ~[reactor-core-3.3.0.RELEASE.jar:3.3.0.RELEASE]
	at reactor.core.publisher.FluxOnErrorResume$ResumeSubscriber.onError(FluxOnErrorResume.java:88) ~[reactor-core-3.3.0.RELEASE.jar:3.3.0.RELEASE]
	at reactor.core.publisher.MonoFlatMapMany$FlatMapManyInner.onError(MonoFlatMapMany.java:247) ~[reactor-core-3.3.0.RELEASE.jar:3.3.0.RELEASE]
	at com.mongodb.reactivestreams.client.internal.SingleResultObservableToPublisher$1.onError(SingleResultObservableToPublisher.java:50) ~[mongodb-driver-reactivestreams-1.12.0.jar:na]
	at com.mongodb.reactivestreams.client.internal.ObservableToPublisher$1.onError(ObservableToPublisher.java:71) ~[mongodb-driver-reactivestreams-1.12.0.jar:na]
	at com.mongodb.async.client.AbstractSubscription.onError(AbstractSubscription.java:129) ~[mongodb-driver-async-3.11.1.jar:na]
	at com.mongodb.async.client.SingleResultCallbackSubscription$1.onResult(SingleResultCallbackSubscription.java:42) ~[mongodb-driver-async-3.11.1.jar:na]
	at com.mongodb.async.client.MongoCollectionImpl$7.onResult(MongoCollectionImpl.java:921) ~[mongodb-driver-async-3.11.1.jar:na]
	at com.mongodb.async.client.MongoCollectionImpl$7.onResult(MongoCollectionImpl.java:917) ~[mongodb-driver-async-3.11.1.jar:na]
	at com.mongodb.async.client.MongoCollectionImpl$9.onResult(MongoCollectionImpl.java:980) ~[mongodb-driver-async-3.11.1.jar:na]
	at com.mongodb.async.client.MongoCollectionImpl$9.onResult(MongoCollectionImpl.java:976) ~[mongodb-driver-async-3.11.1.jar:na]
	at com.mongodb.internal.async.ErrorHandlingResultCallback.onResult(ErrorHandlingResultCallback.java:49) ~[mongodb-driver-core-3.11.1.jar:na]
	at com.mongodb.async.client.OperationExecutorImpl$2$1$1.onResult(OperationExecutorImpl.java:140) ~[mongodb-driver-async-3.11.1.jar:na]
	at com.mongodb.internal.async.ErrorHandlingResultCallback.onResult(ErrorHandlingResultCallback.java:49) ~[mongodb-driver-core-3.11.1.jar:na]
	at com.mongodb.operation.CreateIndexesOperation$2.call(CreateIndexesOperation.java:194) ~[mongodb-driver-core-3.11.1.jar:na]
	at com.mongodb.operation.OperationHelper$AsyncCallableWithConnectionCallback.onResult(OperationHelper.java:569) ~[mongodb-driver-core-3.11.1.jar:na]
	at com.mongodb.operation.OperationHelper$AsyncCallableWithConnectionCallback.onResult(OperationHelper.java:561) ~[mongodb-driver-core-3.11.1.jar:na]
	at com.mongodb.internal.async.ErrorHandlingResultCallback.onResult(ErrorHandlingResultCallback.java:49) ~[mongodb-driver-core-3.11.1.jar:na]
	at com.mongodb.binding.AsyncClusterBinding$1.onResult(AsyncClusterBinding.java:111) ~[mongodb-driver-core-3.11.1.jar:na]
	at com.mongodb.binding.AsyncClusterBinding$1.onResult(AsyncClusterBinding.java:107) ~[mongodb-driver-core-3.11.1.jar:na]
	at com.mongodb.internal.connection.BaseCluster$ServerSelectionRequest.onResult(BaseCluster.java:440) ~[mongodb-driver-core-3.11.1.jar:na]
	at com.mongodb.internal.connection.BaseCluster$WaitQueueHandler.run(BaseCluster.java:512) ~[mongodb-driver-core-3.11.1.jar:na]
	at java.base/java.lang.Thread.run(Thread.java:844) ~[na:na]
Caused by: com.mongodb.MongoClientException: Shutdown in progress
	... 2 common frames omitted

 2020-04-04 18:25:59.567 ERROR  28064 --- [cluster-5e88d1173ea5cc3c08d2aec1] org.mongodb.driver.client                : Callback onResult call produced an error

com.mongodb.MongoClientException: Shutdown in progress
	at com.mongodb.internal.connection.BaseCluster$WaitQueueHandler.run(BaseCluster.java:512) [mongodb-driver-core-3.11.1.jar:na]
	at java.base/java.lang.Thread.run(Thread.java:844) [na:na]

 2020-04-04 18:26:01.722  INFO  28064 --- [restartedMain] o.apache.catalina.core.StandardService   : Stopping service [Tomcat]
 2020-04-04 18:26:01.733  INFO  28064 --- [restartedMain] ConditionEvaluationReportLoggingListener : 

Error starting ApplicationContext. To display the conditions report re-run your application with 'debug' enabled.
 2020-04-04 18:26:01.740 ERROR  28064 --- [restartedMain] o.s.boot.SpringApplication               : Application run failed

org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'mongoTemplate' defined in class path resource [org/springframework/boot/autoconfigure/data/mongo/MongoDbFactoryDependentConfiguration.class]: Bean instantiation via factory method failed; nested exception is org.springframework.beans.BeanInstantiationException: Failed to instantiate [org.springframework.data.mongodb.core.MongoTemplate]: Factory method 'mongoTemplate' threw exception; nested exception is org.springframework.dao.DataAccessResourceFailureException: Timed out after 30000 ms while waiting to connect. Client view of cluster state is {type=UNKNOWN, servers=[{address=localhost:27017, type=UNKNOWN, state=CONNECTING, exception={com.mongodb.MongoSocketOpenException: Exception opening socket}, caused by {java.net.ConnectException: Connection refused (Connection refused)}}]; nested exception is com.mongodb.MongoTimeoutException: Timed out after 30000 ms while waiting to connect. Client view of cluster state is {type=UNKNOWN, servers=[{address=localhost:27017, type=UNKNOWN, state=CONNECTING, exception={com.mongodb.MongoSocketOpenException: Exception opening socket}, caused by {java.net.ConnectException: Connection refused (Connection refused)}}]
	at org.springframework.beans.factory.support.ConstructorResolver.instantiate(ConstructorResolver.java:645) ~[spring-beans-5.2.0.RELEASE.jar:5.2.0.RELEASE]
	at org.springframework.beans.factory.support.ConstructorResolver.instantiateUsingFactoryMethod(ConstructorResolver.java:625) ~[spring-beans-5.2.0.RELEASE.jar:5.2.0.RELEASE]
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.instantiateUsingFactoryMethod(AbstractAutowireCapableBeanFactory.java:1338) ~[spring-beans-5.2.0.RELEASE.jar:5.2.0.RELEASE]
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBeanInstance(AbstractAutowireCapableBeanFactory.java:1177) ~[spring-beans-5.2.0.RELEASE.jar:5.2.0.RELEASE]
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.doCreateBean(AbstractAutowireCapableBeanFactory.java:557) ~[spring-beans-5.2.0.RELEASE.jar:5.2.0.RELEASE]
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBean(AbstractAutowireCapableBeanFactory.java:517) ~[spring-beans-5.2.0.RELEASE.jar:5.2.0.RELEASE]
	at org.springframework.beans.factory.support.AbstractBeanFactory.lambda$doGetBean$0(AbstractBeanFactory.java:323) ~[spring-beans-5.2.0.RELEASE.jar:5.2.0.RELEASE]
	at org.springframework.beans.factory.support.DefaultSingletonBeanRegistry.getSingleton(DefaultSingletonBeanRegistry.java:222) ~[spring-beans-5.2.0.RELEASE.jar:5.2.0.RELEASE]
	at org.springframework.beans.factory.support.AbstractBeanFactory.doGetBean(AbstractBeanFactory.java:321) ~[spring-beans-5.2.0.RELEASE.jar:5.2.0.RELEASE]
	at org.springframework.beans.factory.support.AbstractBeanFactory.getBean(AbstractBeanFactory.java:202) ~[spring-beans-5.2.0.RELEASE.jar:5.2.0.RELEASE]
	at org.springframework.beans.factory.support.DefaultListableBeanFactory.preInstantiateSingletons(DefaultListableBeanFactory.java:879) ~[spring-beans-5.2.0.RELEASE.jar:5.2.0.RELEASE]
	at org.springframework.context.support.AbstractApplicationContext.finishBeanFactoryInitialization(AbstractApplicationContext.java:878) ~[spring-context-5.2.0.RELEASE.jar:5.2.0.RELEASE]
	at org.springframework.context.support.AbstractApplicationContext.refresh(AbstractApplicationContext.java:550) ~[spring-context-5.2.0.RELEASE.jar:5.2.0.RELEASE]
	at org.springframework.boot.web.servlet.context.ServletWebServerApplicationContext.refresh(ServletWebServerApplicationContext.java:141) ~[spring-boot-2.2.0.RELEASE.jar:2.2.0.RELEASE]
	at org.springframework.boot.SpringApplication.refresh(SpringApplication.java:747) [spring-boot-2.2.0.RELEASE.jar:2.2.0.RELEASE]
	at org.springframework.boot.SpringApplication.refreshContext(SpringApplication.java:397) [spring-boot-2.2.0.RELEASE.jar:2.2.0.RELEASE]
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:315) [spring-boot-2.2.0.RELEASE.jar:2.2.0.RELEASE]
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:1226) [spring-boot-2.2.0.RELEASE.jar:2.2.0.RELEASE]
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:1215) [spring-boot-2.2.0.RELEASE.jar:2.2.0.RELEASE]
	at application.MessageApplication.main(MessageApplication.java:14) [classes/:na]
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[na:na]
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[na:na]
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[na:na]
	at java.base/java.lang.reflect.Method.invoke(Method.java:564) ~[na:na]
	at org.springframework.boot.devtools.restart.RestartLauncher.run(RestartLauncher.java:49) [spring-boot-devtools-2.2.0.RELEASE.jar:2.2.0.RELEASE]
Caused by: org.springframework.beans.BeanInstantiationException: Failed to instantiate [org.springframework.data.mongodb.core.MongoTemplate]: Factory method 'mongoTemplate' threw exception; nested exception is org.springframework.dao.DataAccessResourceFailureException: Timed out after 30000 ms while waiting to connect. Client view of cluster state is {type=UNKNOWN, servers=[{address=localhost:27017, type=UNKNOWN, state=CONNECTING, exception={com.mongodb.MongoSocketOpenException: Exception opening socket}, caused by {java.net.ConnectException: Connection refused (Connection refused)}}]; nested exception is com.mongodb.MongoTimeoutException: Timed out after 30000 ms while waiting to connect. Client view of cluster state is {type=UNKNOWN, servers=[{address=localhost:27017, type=UNKNOWN, state=CONNECTING, exception={com.mongodb.MongoSocketOpenException: Exception opening socket}, caused by {java.net.ConnectException: Connection refused (Connection refused)}}]
	at org.springframework.beans.factory.support.SimpleInstantiationStrategy.instantiate(SimpleInstantiationStrategy.java:185) ~[spring-beans-5.2.0.RELEASE.jar:5.2.0.RELEASE]
	at org.springframework.beans.factory.support.ConstructorResolver.instantiate(ConstructorResolver.java:640) ~[spring-beans-5.2.0.RELEASE.jar:5.2.0.RELEASE]
	... 24 common frames omitted
Caused by: org.springframework.dao.DataAccessResourceFailureException: Timed out after 30000 ms while waiting to connect. Client view of cluster state is {type=UNKNOWN, servers=[{address=localhost:27017, type=UNKNOWN, state=CONNECTING, exception={com.mongodb.MongoSocketOpenException: Exception opening socket}, caused by {java.net.ConnectException: Connection refused (Connection refused)}}]; nested exception is com.mongodb.MongoTimeoutException: Timed out after 30000 ms while waiting to connect. Client view of cluster state is {type=UNKNOWN, servers=[{address=localhost:27017, type=UNKNOWN, state=CONNECTING, exception={com.mongodb.MongoSocketOpenException: Exception opening socket}, caused by {java.net.ConnectException: Connection refused (Connection refused)}}]
	at org.springframework.data.mongodb.core.MongoExceptionTranslator.translateExceptionIfPossible(MongoExceptionTranslator.java:90) ~[spring-data-mongodb-2.2.0.RELEASE.jar:2.2.0.RELEASE]
	at org.springframework.data.mongodb.core.MongoTemplate.potentiallyConvertRuntimeException(MongoTemplate.java:2902) ~[spring-data-mongodb-2.2.0.RELEASE.jar:2.2.0.RELEASE]
	at org.springframework.data.mongodb.core.MongoTemplate.execute(MongoTemplate.java:587) ~[spring-data-mongodb-2.2.0.RELEASE.jar:2.2.0.RELEASE]
	at org.springframework.data.mongodb.core.DefaultIndexOperations.execute(DefaultIndexOperations.java:214) ~[spring-data-mongodb-2.2.0.RELEASE.jar:2.2.0.RELEASE]
	at org.springframework.data.mongodb.core.DefaultIndexOperations.ensureIndex(DefaultIndexOperations.java:121) ~[spring-data-mongodb-2.2.0.RELEASE.jar:2.2.0.RELEASE]
	at org.springframework.data.mongodb.core.index.MongoPersistentEntityIndexCreator.createIndex(MongoPersistentEntityIndexCreator.java:158) ~[spring-data-mongodb-2.2.0.RELEASE.jar:2.2.0.RELEASE]
	at org.springframework.data.mongodb.core.index.MongoPersistentEntityIndexCreator.checkForAndCreateIndexes(MongoPersistentEntityIndexCreator.java:148) ~[spring-data-mongodb-2.2.0.RELEASE.jar:2.2.0.RELEASE]
	at org.springframework.data.mongodb.core.index.MongoPersistentEntityIndexCreator.checkForIndexes(MongoPersistentEntityIndexCreator.java:130) ~[spring-data-mongodb-2.2.0.RELEASE.jar:2.2.0.RELEASE]
	at org.springframework.data.mongodb.core.index.MongoPersistentEntityIndexCreator.<init>(MongoPersistentEntityIndexCreator.java:95) ~[spring-data-mongodb-2.2.0.RELEASE.jar:2.2.0.RELEASE]
	at org.springframework.data.mongodb.core.index.MongoPersistentEntityIndexCreator.<init>(MongoPersistentEntityIndexCreator.java:72) ~[spring-data-mongodb-2.2.0.RELEASE.jar:2.2.0.RELEASE]
	at org.springframework.data.mongodb.core.MongoTemplate.<init>(MongoTemplate.java:275) ~[spring-data-mongodb-2.2.0.RELEASE.jar:2.2.0.RELEASE]
	at org.springframework.boot.autoconfigure.data.mongo.MongoDbFactoryDependentConfiguration.mongoTemplate(MongoDbFactoryDependentConfiguration.java:63) ~[spring-boot-autoconfigure-2.2.0.RELEASE.jar:2.2.0.RELEASE]
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[na:na]
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[na:na]
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[na:na]
	at java.base/java.lang.reflect.Method.invoke(Method.java:564) ~[na:na]
	at org.springframework.beans.factory.support.SimpleInstantiationStrategy.instantiate(SimpleInstantiationStrategy.java:154) ~[spring-beans-5.2.0.RELEASE.jar:5.2.0.RELEASE]
	... 25 common frames omitted
Caused by: com.mongodb.MongoTimeoutException: Timed out after 30000 ms while waiting to connect. Client view of cluster state is {type=UNKNOWN, servers=[{address=localhost:27017, type=UNKNOWN, state=CONNECTING, exception={com.mongodb.MongoSocketOpenException: Exception opening socket}, caused by {java.net.ConnectException: Connection refused (Connection refused)}}]
	at com.mongodb.internal.connection.BaseCluster.getDescription(BaseCluster.java:182) ~[mongodb-driver-core-3.11.1.jar:na]
	at com.mongodb.internal.connection.AbstractMultiServerCluster.getDescription(AbstractMultiServerCluster.java:54) ~[mongodb-driver-core-3.11.1.jar:na]
	at com.mongodb.client.internal.MongoClientDelegate.getConnectedClusterDescription(MongoClientDelegate.java:145) ~[mongodb-driver-3.11.1.jar:na]
	at com.mongodb.client.internal.MongoClientDelegate.createClientSession(MongoClientDelegate.java:100) ~[mongodb-driver-3.11.1.jar:na]
	at com.mongodb.client.internal.MongoClientDelegate$DelegateOperationExecutor.getClientSession(MongoClientDelegate.java:277) ~[mongodb-driver-3.11.1.jar:na]
	at com.mongodb.client.internal.MongoClientDelegate$DelegateOperationExecutor.execute(MongoClientDelegate.java:201) ~[mongodb-driver-3.11.1.jar:na]
	at com.mongodb.client.internal.MongoCollectionImpl.executeCreateIndexes(MongoCollectionImpl.java:886) ~[mongodb-driver-3.11.1.jar:na]
	at com.mongodb.client.internal.MongoCollectionImpl.createIndexes(MongoCollectionImpl.java:869) ~[mongodb-driver-3.11.1.jar:na]
	at com.mongodb.client.internal.MongoCollectionImpl.createIndexes(MongoCollectionImpl.java:864) ~[mongodb-driver-3.11.1.jar:na]
	at com.mongodb.client.internal.MongoCollectionImpl.createIndex(MongoCollectionImpl.java:849) ~[mongodb-driver-3.11.1.jar:na]
	at org.springframework.data.mongodb.core.DefaultIndexOperations.lambda$ensureIndex$0(DefaultIndexOperations.java:131) ~[spring-data-mongodb-2.2.0.RELEASE.jar:2.2.0.RELEASE]
	at org.springframework.data.mongodb.core.MongoTemplate.execute(MongoTemplate.java:585) ~[spring-data-mongodb-2.2.0.RELEASE.jar:2.2.0.RELEASE]
	... 39 common frames omitted

 